File: Java/src/basedatos/Deserializador.java
package basedatos;

import java.io.EOFException;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.util.ArrayList;
import java.util.List;

import gestorAplicacion.infoClientes.Banco;
import gestorAplicacion.infoClientes.PerfilCreditico;
import gestorAplicacion.infoClientes.Transaccion;
import gestorAplicacion.infoClientes.Usuario;

import gestorAplicacion.productosFinancieros.Ahorro;
import gestorAplicacion.productosFinancieros.BajoMonto;
import gestorAplicacion.productosFinancieros.Corriente;
import gestorAplicacion.productosFinancieros.Credito;


/**
 * Clase para deserializar los objetos que se crearon en ejecucion
 * @author Erik Gonzalez
 * @author Felipe Miranda
 */
public class Deserializador {
	/**
	 * Utilizamos clases genericas para permitir reutilizar la funcion para todas
	 * las clases del proyecto
	 * 
	 * @param <E>       el generico se usa para poder agredar las clases que se
	 *                  crearon
	 * @param lista     Una lista de objetos
	 * @param className El nombre de la clase que queremos usar como nombre del
	 *                  archivo
	 */
	public static <E> void deserializador(List<E> list, String className) {
		FileInputStream fileIn;
		try {
			// Creamos una cadena con la ruta del archivo que vamos a cargar
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
//			System.out.println(path);
			// utilizamos un file para crear este archivo si no existe aun
			File archivo = new File(path);
			archivo.createNewFile(); // Crea un nuevo archivo si no existe

			// usamos un FileInputStream para poder saber de donde cargar el archivo
			fileIn = new FileInputStream(path);

			// Si el archivo esta vacio se lanza un throw EOFException y se muestra como un
			// mensaje de vacio, pero si no se usa el objeto in para leer el archivo
			ObjectInputStream in = new ObjectInputStream(fileIn);

			// Lee el listado de elementos
			ArrayList<E> listado = (ArrayList<E>) in.readObject();

			// Recorro el ArrayList
			for (E el : listado) {
				list.add(el);
			}

			in.close();
			fileIn.close();

		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			System.out.println("Esta vacio");
		} catch (ClassNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

	/**
	 * Funcion para deserializar toda la aplicacion Generic IT
	 */
	public static void deserializarTodo() {
		Deserializador.deserializador(Banco.listaBancos, "Banco");
		Deserializador.deserializador(PerfilCreditico.getPerfilCreditico(), "PerfilCreditico");
		Deserializador.deserializador(Transaccion.getTransaccion(), "Transaccion");
		Deserializador.deserializador(Usuario.getUsuario(), "Usuario");
		Deserializador.deserializador(Ahorro.getAhorro(), "Ahorro");
		Deserializador.deserializador(BajoMonto.getBajoMonto(), "BajoMonto");
		Deserializador.deserializador(Corriente.getCorriente(), "Corriente");
		Deserializador.deserializador(Credito.getCredito(), "Credito");
	}
}

File: Python/src/Wunallet/capaLogica/baseDatos/deserializador.py
'''Este módulo fue hecho con la finalidad de crear la función para deserializar los objetos guardados, para
así luego  usados en el main de nuestra aplicación.'''

import os
import pathlib
import pickle

from Wunallet.capaLogica.gestorAplicacion.infoClientes.banco import Banco
from Wunallet.capaLogica.gestorAplicacion.infoClientes.perfilCrediticio import PerfilCrediticio
from Wunallet.capaLogica.gestorAplicacion.infoClientes.transaccion import Transaccion
from Wunallet.capaLogica.gestorAplicacion.infoClientes.usuario import Usuario

from Wunallet.capaLogica.gestorAplicacion.productosFinancieros.ahorro import Ahorro
from Wunallet.capaLogica.gestorAplicacion.productosFinancieros.bajoMonto import BajoMonto
from Wunallet.capaLogica.gestorAplicacion.productosFinancieros.corriente import Corriente
from Wunallet.capaLogica.gestorAplicacion.productosFinancieros.credito import Credito

class Deserializador():
    print("Entré a deserializar")
    def deserializar(lista, className):
        def camino(className):
            string = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\"+className+".txt")
            print("La ruta donde estoy buscando", string)
            return string
        # Leo el archivo
        try:
            picklefile = open(camino(className), 'rb')
        except:
            picklefile = open(camino(className), 'x')
            picklefile = open(camino(className), 'rb')
        # unpickle los datos
        if os.path.getsize(camino(className)) > 0:
            lista = pickle.load(picklefile)
        
        # Cierro el archivo
        picklefile.close()
        print("Si funcioné", lista)
        return lista
        # Cierro el archivo
    
    def deserializarTodo():
        Banco.banco = Deserializador.deserializar(Banco.banco, "Banco")
        PerfilCrediticio._perfilCrediticio = Deserializador.deserializar(PerfilCrediticio._perfilCrediticio, "PerfilCrediticio")
        Transaccion._transaccion =  Deserializador.deserializar(Transaccion._transaccion, "Transaccion")
        Usuario._usuario = Deserializador.deserializar(Usuario._usuario, "Usuario")
        Ahorro.ahorro = Deserializador.deserializar(Ahorro.ahorro, "Ahorro")
        BajoMonto._bajoMonto = Deserializador.deserializar(BajoMonto._bajoMonto, "BajoMonto")
        Corriente._corriente = Deserializador.deserializar(Corriente._corriente, "Corriente")
        Credito._credito = Deserializador.deserializar(Credito._credito, "Credito")


        # b = Deserializador.deserializar(Banco.banco, "Banco")
        # Banco.setBanco(b)

        # # p = Deserializador.deserializar(PerfilCrediticio._perfilCrediticio, "PerfilCrediticio")
        # # PerfilCrediticio.setPerfilCrediticio(p)
        
        # t =  Deserializador.deserializar(Transaccion._transaccion, "Transaccion")
        # Transaccion.setTransaccion(t)
        # print("SI LO ESTOY HACIENDO",t)

        # u = Deserializador.deserializar(Usuario._usuario, "Usuario")
        # Usuario.setUsuario(u)

        # a = Deserializador.deserializar(Ahorro.ahorro, "Ahorro")
        # Ahorro.setAhorro(a)

        # ba = Deserializador.deserializar(BajoMonto._bajoMonto, "BajoMonto")
        # BajoMonto.setBajoMonto(ba)

        # co = Deserializador.deserializar(Corriente._corriente, "Corriente")
        # Corriente.setCorriente(co)

        # cre = Deserializador.deserializar(Credito._credito, "Credito")
        # Credito.setCredito(cre)
